src/app/clusters/icd-management-server/icd-management-server.cpp

```diff
@@ -107,13 +107,14 @@ CHIP_ERROR IcdManagementAttributeAccess::ReadRegisteredClients(EndpointId endpoi
     uint16_t supported_clients = IcdManagementServer::GetInstance().GetClientsSupportedPerFabric();
 
     return encoder.EncodeList([supported_clients](const auto & subEncoder) -> CHIP_ERROR {
-        IcdMonitoringEntry e;
+        IcdMonitoringEntry e(chip::Server::GetInstance().GetSessionKeystore());
```

Comment:
Can you create an issue so we can have a follow up on removing a dependency on `Server` here? 
Same as the PersistantStorage object

Reply:
#29289